### PR TITLE
feat: support force delete resource

### LINKED ISSUE #9795: proposal: support force delete resource without reference check
### Current State

Currently, some resource deletion will perform a reference check. It is good for manual operations, but not for automated situations designed to meet an eventual consistency.

### Desired State

Add an option to force delete resources.

### PR DESCRIPTION
**Pull request by lingsamuel**
### Description

Support to force delete resources with header `X-Force-Delete: true`.

Fixes #9795

example:

```bash
❯ curl http://127.0.0.1:9180/apisix/admin/upstreams/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '{
    "nodes": {
        "127.0.0.1:8080": 1
    },
    "type": "roundrobin"
}'
❯ curl http://127.0.0.1:9180/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '{
    "uri": "/*",
    "upstream_id": 1
}'
{"value":{"priority":0,"upstream_id":1,"uri":"/*","create_time":1689038794,"id":"1","status":1,"update_time":1689038916},"key":"/apisix/routes/1"}

❯ curl http://127.0.0.1:9180/apisix/admin/upstreams/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X DELETE
{"error_msg":"can not delete this upstream, route [1] is still using it now"}
❯ curl http://127.0.0.1:9180/apisix/admin/upstreams/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X DELETE -H 'X-Force-Delete: anyvalue'
{"error_msg":"can not delete this upstream, route [1] is still using it now"}
❯ curl http://127.0.0.1:9180/apisix/admin/upstreams/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X DELETE -H 'X-Force-Delete: true'
{"deleted":"1","key":"/apisix/upstreams/1"}
```

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/admin-api.md [hunk 1]
```diff
@@ -103,6 +103,33 @@ deployment:
 
 This will find the environment variable `ADMIN_KEY` first, and if it does not exist, it will use `edd1c9f034335f136f87ad84b625c8f1` as the default value.
 
+### Force Delete
+
+By default, the Admin API checks for references between resources and will refuse to delete resources in use.
+
+You can make a force deletion by adding the request header `X-Force-Delete: true` to the delete request, for example:
```
Comment (monkeyDluffy6017) [inline docs/en/latest/admin-api.md:110]: Is it better to put this in `uri_args` ?

Comment (lingsamuel) [reply docs/en/latest/admin-api.md:110]: I have no preference for this design. wdyt @kingluo @leslie-tsang

Comment (monkeyDluffy6017) [reply docs/en/latest/admin-api.md:110]: If you put the configuration into the uri_args, then you don't need to add this following code
```
        if method == "delete" then
            if not req_body then
                req_body = {}
            end
            req_body._force_delete = ngx.req.get_headers()["X-Force-Delete"]
        end
```

**Comment by kingluo** [comment]
URL arg is better. The admin API did not use arguments, so it's ok to insist this simple way.

File: docs/en/latest/admin-api.md (hunk 1 above)
Comment (leslie-tsang) [reply docs/en/latest/admin-api.md:110]: I prefer placing the parameter in `uri_args` to keep this interface as simple as possible. :)

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**